lib/client/connection.js

```diff
@@ -48,6 +48,8 @@ function Connection(socket) {
 
   this.debug = false;
 
+  this.state = 'disconnected'
```

Comment:
To avoid emission on initial construction, I'd set this based on the state of the socket passed in:
```
this.state = connectionState(socket);
```

And add the function:
```
function connectionState(socket) {
  if (socket.readyState === 0 || socket.readyState === 1) return 'connecting';
  return 'disconnected';
}
```